### PR TITLE
Remove the "evolve_captured" flag in favor of the EvolveTask

### PR DESCRIPTION
**Pull request by achretien**
Remove the "evolve_captured" flag in favor of the EvolveTask

The current flag don't use the evolve config

### CODE REVIEW

**Comment by viniciusbo** [comment]
I think those changes will prevent pokemons from being evolved when catched.

**Comment by marcosalles** [comment]
that prevention might be a good thing, since the `evolve_captured` filter doesn't go well with your other filters, like keeping the pokemon based on CP or IV stats.

File: pokecli.py [hunk 1]
```diff
@@ -453,11 +444,8 @@ def task_configuration_error(flag_name):
             task_configuration_error('{}.{}'.format(outer, inner))
             return None
 
-    if (config.evolve_captured
-        and (not isinstance(config.evolve_captured, str)
-             or str(config.evolve_captured).lower() in ["true", "false"])):
-        parser.error('"evolve_captured" should be list of pokemons: use "all" or "none" to match all ' +
-                     'or none of the pokemons, or use a comma separated list such as "Pidgey,Weedle,Caterpie"')
+    if config.evolve_captured:
+        parser.error('"evolve_captured" has been removed in favor of the EvolveTask')
```
Comment (TheSavior) [inline pokecli.py]: Can we make this a deprecation warning that lets the bot keep running? I'd like to avoid breaking peoples' configs if possible.

**Comment by TheSavior** [comment]
I agree, catching pokemon should have nothing to do with evolving them. If people want to evolve things, the evolve task should be configured as such.

I would like another input before merging this though. @douglascamata?

:+1:

**Comment by douglascamata** [comment]
👍 

YOU SHALL BE MEEEERGED

**Comment by 1cu** [comment]
Just feedback to this merged PR:

evolve_capture were totally independent features and at least in my case are configured differently. 

i evolve **all** lowlevel pkmn on capture but the higher ones are evolved based on certain criteria (cp/iv). 

evolve_capture can't be substituted with EvolvePokemon since i can't specify different sets with different criteria :-(

:-1:

**Comment by vicch** [comment]
> catching pokemon should have nothing to do with evolving them. If people want to evolve things, the evolve task should be configured as such.

But `EvolvePokemon` cannot do "catch & evolve", it only evolves at startup.

**Comment by alexyaoyang** [comment]
I agree with @1cu and @vicch. Evolving some mons (such as weedle, caterpie, pidgey) when catching allows increase in exp/hr and I don't really care about keeping any of those evolves. Evolve captured is a independent feature from EvolveTask.

**Comment by achretien** [comment]
You can still add a second task evolve after the catch one.

**Comment by vicch** [comment]
You mean like this?
```
{
    "type": "CatchVisiblePokemon",
    "config": { "enabled": true }
},
{
    "type": "CatchLuredPokemon",
    "config": { "enabled": true }
},
{
    "type": "EvolvePokemon",
    "config": {
        "enabled": true,
        "evolve_all": "Caterpie,Weedle,Pidgey",
        "first_evolve_by": "cp",
        "evolve_above_cp": 0,
        "evolve_above_iv": 0,
        "logic": "or",
        "evolve_speed": 12,
        "use_lucky_egg": false
    }
},
```
But it doesn't seem to work as I tried.

**Comment by achretien** [comment]
I have to check and will send you my config if you want.

I have only one evolve task run at each tick

**Comment by vicch** [comment]
If you could share part of your config that would be great, thanks.
My feeling is that `EvolvePokemon` interferes with `TransferPokemon`, which makes it not work as wished.

**Comment by achretien** [comment]
>
>  I will


I have the transfer task after the evolve one to evolve before transfer.

**Comment by douglascamata** [comment]
Proper configuration will evolve pokemon before being captured. We don't need duplicated logic.

**Comment by achretien** [comment]
@vicch

```

    {
      "type": "EvolvePokemon",
      "config": {
        "evolve_all": "Staryu,Goldeen,Doduo,Bellsprout,Psyduck,Krabby,Drowzee,Slowpoke,Eevee,Venonat,Exeggcute,Paras,Ekans,Caterpie,Weedle,Pidgey,Rattata,Spearow,Zubat,Poliwhirl",
        "first_evolve_by": "iv",
        "evolve_above_cp": 0,
        "evolve_above_iv": 0,
        "logic": "and",
        "evolve_speed": 1,
        "use_lucky_egg": false
      }
    },
    {
      "type": "TransferPokemon"
    },
    {
      "type": "MoveToFort",
      "config": {
        "ignore_item_count": true,
        "lure_attraction": true,
        "lure_max_distance": 2000
      }
    },
    {
      "type": "SpinFort",
      "config": {
        "ignore_item_count": true
      }
    },
    {
      "type": "CatchVisiblePokemon"
    },
    {
      "type": "CatchLuredPokemon"
    }
```

**Comment by vicch** [comment]
I tried again
```
{
    "type": "CatchVisiblePokemon",
    "config": { "enabled": true }
},
{
    "type": "CatchLuredPokemon",
    "config": { "enabled": true }
},
{
    "type": "EvolvePokemon",
    "config": {
        "enabled": true,
        "evolve_all": "Caterpie,Weedle,Pidgey,Rattata,Spearow,Zubat,Magnemite,Voltorb",
        "first_evolve_by": "cp",
        "evolve_above_cp": 0,
        "evolve_above_iv": 0,
        "logic": "or",
        "evolve_speed": 12,
        "use_lucky_egg": false
    }
},
{
    "type": "TransferPokemon",
    "config": { "enabled": true }
},
```

But still
```
[12:43:04] Captured Weedle! [CP 180] [Potential 0.44] [1/15/4] [+210 exp]
[12:43:04] You now have 41 Weedle candy!
[12:43:09] Releasing Weedle [CP 180] [IV 0.44] based on rule: CP < 390 AND IV < 0.8
[12:43:10] Exchanged Weedle [CP 180] [IV 0.44] for candy.
[12:43:14] Spun pokestop General Sherman Sculpture. Experience awarded: 50. Items awarded: ...
```

Although it actually worked for a very few of them
```
2016-08-11 12:43:30,328 [       cli] [INFO] Encountered 178 pokemon, 175 caught, 174 released, 4 evolved, 0 never seen before
```

**Comment by vicch** [comment]
Is the `evolve_speed`'s problem? I will try again with `1`.

**Comment by achretien** [comment]
The evolve task don’t refresh the inventory by default.

I think the catch tasks don’t use the invenotory class and don’t add the new catch pokemon to the bag.

 

So we need to refresh the inventory before each resolve task or use the inventory class in the catch classes to update the inventory in fnction of actions (berry, balls and pokemon number calculation)

**Comment by douglascamata** [comment]
@achretien it is working already, see:

```
2016-08-11 21:51:40,990 [PokemonCatchWorker] [INFO] [pokemon_appeared] A wild Pidgey appeared! [CP 10] [Potential 0.47] [A/D/S 5/15/1]
2016-08-11 21:51:41,637 [PokemonCatchWorker] [INFO] [threw_pokeball] Used Pokeball, with chance 100.00 (52 left)
2016-08-11 21:51:42,232 [PokemonCatchWorker] [INFO] [pokemon_caught] Captured Pidgey! [CP 10] [Potential 0.47] [5/15/1] [+210 exp]
2016-08-11 21:51:42,232 [PokemonCatchWorker] [INFO] [gained_candy] You now have 14 Pidgey candy!
2016-08-11 21:51:43,555 [EvolvePokemon] [INFO] [pokemon_evolved] Successfully evolved Pidgey with CP 25 and IV 0.42!```

**Comment by douglascamata** [comment]
You just need to have EvolvePokemon in the beginning of the tick, before release and before capture.

**Comment by vicch** [comment]
Last attempt, I added 4 `EvolvePokemon` tasks:
1. The first task
2. Directly before `TransferPokemon`
3. Directly after `TransferPokemon`
4. The last task

Each with config
```
{
    "type": "EvolvePokemon",
    "config": {
        "enabled": true,
        "evolve_all": "Caterpie,Weedle,Pidgey,Rattata,Spearow,Zubat,Magnemite,Voltorb",
        "first_evolve_by": "cp",
        "evolve_above_cp": 1,
        "evolve_above_iv": 0.01,
        "logic": "or",
        "evolve_speed": 11,
        "use_lucky_egg": false
    }
},
```
And still not evolving after capture, I give up :joy:

**Comment by douglascamata** [comment]
@vicch task order:

1. EvolvePokemon
2. ReleasePokemon
3. CatchPokemon

It will work.

**Comment by achretien** [comment]
I check earlier and I am pretty sure than the capture task does'nt add the
captured Pokemon to the inventory.

I will add a PR to add this part. The new inventory part is great and will
reduce the number of calls to the API. But we need to add it in all the
tasks using the inventory.

**Comment by vicch** [comment]
@douglascamata Guess what? Still the same. I'm cursed...
@achretien So it lacks some logic for it to work properly. But it's curious why it works for all of you, but not me.

**Comment by viniciusbo** [comment]
Not working for me either.
Em qui, 11 de ago de 2016 às 17:53, Victor Chen <notifications@github.com>
escreveu:

**Pull request merged**